### PR TITLE
[Performance Page] Meshery UI restructuring

### PR DESCRIPTION
**Pull request by kayceeDev**
**Description**

This PR Adds a static page for the UI performance page

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: ui/components/ExpansionPanels.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Accordion, AccordionSummary } from "@mui/material";
+import { withStyles } from "@mui/styles";
+
+export const ExpansionPanel = withStyles({
+  root: { border: "1px solid rgba(0,0,0,.125)" },
+  expanded: { margin: "auto" },
+})(Accordion);
+
+export const ExpansionPanelSummary = withStyles({
+  root: { borderBottom: "1px solid rgba(0,0,0,.125)" },
```
Comment (leecalcote) [inline ui/components/ExpansionPanels.js:10]: Danger, Will Robinson!

@meshery/meshery-ui-maintainers, should we be avoiding `root` styling?

Comment (Nikhil-Ladha) [reply ui/components/ExpansionPanels.js:10]: Haha, someone watched _Lost in Space_ huh?

Coming to fhe question, I think based in how mui works this will be limited to the div that the component creates. But, will let others confirm...

**Review by YashKamboj — CHANGES_REQUESTED** [review]
@kayceeDev We are no longer using makestyles please use styled component like in the other PRs

File: ui/components/ExpansionPanels.js (hunk 1 above)
Comment (YashKamboj) [reply ui/components/ExpansionPanels.js:10]: we are no longer using make styles and the styles added here in the withstyles can be added in the theme.js file

Comment (Abhishek-kumar09) [reply ui/components/ExpansionPanels.js:10]: came out that root styling was not the problem on later work. So its fine to use "root" styling.

**Comment by kayceeDev** [comment]
> @kayceeDev We are no longer using makestyles please use styled component like in the other PRs

Alright. Thanks. Adjustments will be made.

File: ui/components/MesheryPerformance/index.js [hunk 2]
```diff
@@ -0,0 +1,1039 @@
+//@ts-check
```
Comment (YashKamboj) [inline ui/components/MesheryPerformance/index.js:1]: @kayceeDev I have already made this file and this is merged, please have another look at that.

File: ui/components/GenericModal.jsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+// @ts-nocheck
```
Comment (YashKamboj) [inline ui/components/GenericModal.jsx:1]: @kayceeDev We already have this file

File: ui/components/Modal.js [hunk 4]
```diff
@@ -3,10 +3,10 @@ import { styled } from "@mui/material/styles";
 import { useTheme } from "@mui/system";
 
 const CustomModalWrapper = styled(Modal)(({ theme }) => ({
-   display : "flex",
-   alignItems : "center",
-   justifyContent : "center", 
-}))
+  display: "flex",
```
Comment (YashKamboj) [inline ui/components/Modal.js:6]: unwanted changes

File: ui/.gitignore [hunk 5]
```diff
@@ -1,6 +1,12 @@
-# Cypress Tests Files
-cypress/videos
-cypress/screenshots
-# Code coverage
-coverage
-.nyc_output
+# Cypress Tests Files
+cypress/videos
```
Comment (YashKamboj) [inline ui/.gitignore:2]: Please  revert these changes

File: ui/components/ExpansionPanels.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { Accordion, AccordionSummary } from "@mui/material";
```
Comment (YashKamboj) [inline ui/components/ExpansionPanels.js:1]: You can make these changes directly in the theme.js file

File: ui/components/GrafanaCustomChart.js [hunk 7]
```diff
@@ -0,0 +1,820 @@
+import { Component, useEffect } from "react";
+import PropTypes from "prop-types";
+import { withStyles } from "@mui/styles";
```
Comment (YashKamboj) [inline ui/components/GrafanaCustomChart.js:3]: we are not using  withstyles anymore

File: ui/components/GrafanaCustomCharts.js [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+    marginTop : "1rem",
+  },
+  icon : { width : theme.spacing(2.5), },
+  dialogTitle : { '&>*' : { display : 'flex',
+    alignItems : 'center',
+    justifyContent : 'space-between', } },
+});
+
+
+class GrafanaCustomCharts extends Component {
```
Comment (YashKamboj) [inline ui/components/GrafanaCustomCharts.js:35]: we are not using  the class component anymore make functional component

File: ui/components/GrafanaCustomGaugeChart.js [hunk 9]
```diff
@@ -0,0 +1,121 @@
+import { NoSsr } from '@mui/material';
+import {makeStyles} from '@mui/styles';
```
Comment (YashKamboj) [inline ui/components/GrafanaCustomGaugeChart.js:2]: not using makestyles anymore

File: ui/components/MesheryChart.js [hunk 10]
```diff
@@ -1,50 +1,50 @@
...
-      <Grid container spacing={1} style={{ margin : "1rem" }} justifyContent="center">
-            {NonRecursiveConstructDisplayCells(chartData?.options?.metadata )?.map((el, i) => {
-              return <Grid item xs={4} key={`nri-${i}`}>{el}  </Grid>
-            })}
-          </Grid>
-    </div>
-  )
-}
-
+import React from 'react'
```
Comment (YashKamboj) [inline ui/components/MesheryChart.js:1]: there are no changes here, please revert this

File: ui/components/MesheryDateTimePicker.js [hunk 11]
```diff
@@ -0,0 +1,78 @@
+import { DateTimePicker } from '@mui/x-date-pickers';
+import PropTypes from 'prop-types';
+import { Component } from 'react';
+import { TextField } from '@mui/material';
+import moment from 'moment';
+
+class MesheryDateTimePicker extends Component {
```
Comment (YashKamboj) [inline ui/components/MesheryDateTimePicker.js:7]: not using class component anymore

File: ui/yarn.lock [hunk 12]
```diff
@@ -3,34 +3,24 @@
 
 
 "@ampproject/remapping@^2.1.0":
```
Comment (YashKamboj) [inline ui/yarn.lock:5]: don't push changes inside yarn.lock

**Review by YashKamboj — CHANGES_REQUESTED** [review]
@kayceeDev please make these changes

**Comment by leecalcote** [comment]
@kayceeDev are you circling back to this PR?

**Comment by kayceeDev** [comment]
I am definitely.
